Add DataDogStatsdEventLogger

zmmille2 opened the pull request:
Register IEventLogger service
Create IEventLogger interface
Implement IEventLogger interface as DataDogStatsdEventLogger
Show sample for Controllers
Import library for DataDog

zmmille2 committed:
Add DataDogStatsdEventLogger

Register IEventLogger service
Create IEventLogger interface
Implement IEventLogger interface as DataDogStatsdEventLogger
Show sample for Controllers
Import library for DataDog

zmmille2 commented on HelloApi/HelloApi/App_Start/WebApiConfig.cs:
```diff
@@ -11,6 +13,12 @@ public static void Register(HttpConfiguration config)
         {
             // Web API configuration and services
 
+            var statsdConfig = new StatsdConfig() { StatsdServerName = "127.0.0.1", StatsdPort = 8275 };
```
Should I try to read this from config or is that overengineering for this?

zmmille2 replied:
Also todo figure out where to actually send these

tbarlow12 replied:
I think this is fine considering this is just a sample app that we're going to throw away

tbarlow12 commented on HelloApi/HelloApi/App_Start/WebApiConfig.cs:
```diff
@@ -11,6 +13,12 @@ public static void Register(HttpConfiguration config)
             // Web API configuration and services
 
+            var statsdConfig = new StatsdConfig() { StatsdServerName = "127.0.0.1", StatsdPort = 8275 };
+            using (var dogStatsDService = new DataDogStatsdEventLogger(statsdConfig))
```
is this going to be disposed before it can be used

zmmille2 replied:
unfortunately, I think it is. I had better change this.

tbarlow12 commented on HelloApi/HelloApi/HelloApi.csproj:
```diff
@@ -161,6 +164,9 @@
       <DependentUpon>Global.asax</DependentUpon>
     </Compile>
     <Compile Include="Properties\AssemblyInfo.cs" />
+    <Compile Include="Filters\EventFilter.cs" />
+    <Compile Include="Loggers\IEventLogger.cs" />
+    <Compile Include="Loggers\DataDogStatsdEventLogger.cs" />
```
Maybe it's been a bit since I've been in the c# world - do you need to specify every file here? or is there some kind of `*.cs` convention you can use?

zmmille2 replied:
No idea. I haven't worked in visual studio in a while, this was managed for me by the program itself :\

tbarlow12 reviewed (approved):
Added some questions/things to consider. Looks good. Curious why build is failing...

zmmille2 committed:
remove using statement to prevent loss of scope during normal operations
remove public access modifier from IEventLogger interface

tessmichi committed:
moving again

tessmichi committed:
Merge pull request #2 from fireflies-crew/jefftess/vm

migrating old PR to new repo

tbarlow12 committed:
feat: Terratest CI (#26)

- Adds a basic Terratest test (resource count only for now)
- Adds a Dockerfile for creating an image where Terratests can be executed
- Adds Terratest job to `config.yml` for running in CircleCI

zmmille2 committed:
Use Pure DI rather than a DI container

zmmille2 committed:
fix to use local port for datadog agent

tbarlow12 committed:
fix: Update expected resources in unit test (#28)

jding05 committed:
add deployment step readme (#8)

tessmichi committed:
add some security features (#22)

* everything but MI

* stuff

* remove comments

* enable managed identity

* fix unit test

zmmille2 committed:
Add Unity DI to project.
Update registrations to leverage Unity.
Add sample of Log Filter and DI'd logger instance.

zmmille2 committed:
remove MonoDevelop element from .csproj

zmmille2 committed:
add Moq library.
Mock DI for test.

zmmille2 committed:
add Moq reference
